client/signup/steps/page-picker/use-cart-for-difm.tsx

```diff
@@ -0,0 +1,251 @@
+	MinimalRequestCartProduct,
+	ResponseCart,
+	ResponseCartProduct,
+} from '@automattic/shopping-cart';
+import type { ProductListItem } from 'calypso/state/products-list/selectors/get-products-list';
+import type { TranslateOptions, TranslateResult } from 'i18n-calypso';
+
+export type CartItem = {
+	nameOverride?: TranslateResult;
+	product:
+		| ( Partial< ProductListItem > & Pick< ProductListItem, 'cost' | 'product_name' > )
+		| ( Partial< ResponseCartProduct > & Pick< ResponseCartProduct, 'cost' | 'product_name' > );
+	meta?: TranslateResult;
+	productCount?: number;
+	lineCost: number;
+};
+
+const FREE_PAGES = 5;
+
+type TranslateType = ( original: string, options?: TranslateOptions ) => React.ReactChild;
+type DummyCartParams = {
+	selectedPages: string[];
+	currencyCode: string;
+	isPaidPlan: boolean | null;
+	activePlanScheme: ProductListItem | null;
+	difmLiteProduct: ProductListItem | null;
+	extraPageProduct: ProductListItem | null;
+	translate: TranslateType;
+	formatCurrency: ( original: number, currrency: string ) => string | null;
+};
+
+function getDummyCartProducts( {
+	selectedPages,
+	currencyCode,
+	isPaidPlan,
+	activePlanScheme,
+	difmLiteProduct,
+	extraPageProduct,
+	translate,
+	formatCurrency,
+}: DummyCartParams ): CartItem[] {
+	const extraPageCount = Math.max( 0, selectedPages.length - FREE_PAGES );
+	let displayedCartItems: CartItem[] = [];
+	if ( difmLiteProduct && activePlanScheme && extraPageProduct ) {
+		displayedCartItems = [
+			{
+				nameOverride: translate( 'Website Design Service' ),
+				product: difmLiteProduct,
+				lineCost: difmLiteProduct.cost,
+				meta: translate( 'One-time fee' ),
+			},
+			{
+				product: activePlanScheme,
+				meta: translate( 'Plan Subscription: %(planPrice)s per year', {
+					args: { planPrice: formatCurrency( activePlanScheme.cost, currencyCode ) },
+				} ),
+				lineCost: activePlanScheme.cost,
+			},
+
+			{
+				nameOverride: `${ extraPageCount } ${
+					extraPageCount === 1 ? translate( 'Extra Page' ) : translate( 'Extra Pages' )
+				}`,
+				product: extraPageProduct,
+				meta: translate( '%(perPageCost)s Per Page', {
+					args: {
+						perPageCost: extraPageProduct.cost_display,
+					},
+				} ),
+				lineCost: extraPageProduct.cost * extraPageCount,
+				productCount: extraPageCount,
+			},
+		];
+	}
+
+	// Hide pro plan if existing site has a paid plan
+	if ( activePlanScheme && isPaidPlan ) {
+		displayedCartItems = displayedCartItems.filter(
+			( p ) => p.product.product_slug !== activePlanScheme.product_slug
+		);
+	}
+	return displayedCartItems;
+}
+
+function getSiteCartProducts( {
+	responseCart,
+	translate,
+}: {
+	responseCart: ResponseCart;
+	translate: TranslateType;
+} ): CartItem[] {
+	return responseCart.products.map( ( product ) => {
+		switch ( product.product_slug ) {
+			case PLAN_WPCOM_PRO:
+				return {
+					product: product,
+					lineCost: product.cost,
+					meta: translate( 'Plan Subscription: %(planPrice)s per year', {
```

Comment:
I was using a custom label because of the custom suggestions from Lucas for the cart items. Do you think its ok to just show the sub label in checkout.